internal/k8s/k8s.go

```diff
@@ -125,7 +147,13 @@ func New(cfg *Config) (*Client, error) {
 				&corev1.Secret{}:                   namespaceSelector,
 			},
 		}),
+		//todo: check if needed
+		CertDir: cfg.CertDir,
+		MapperProvider: func(c *rest.Config) (meta.RESTMapper, error) {
```

Comment:
This is not needed, as it's equal to the default:
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cluster/cluster.go#L214